consensus/polybft/consensus_runtime.go

```diff
@@ -421,35 +426,32 @@ func (c *consensusRuntime) FSM() error {
 
 	valSet := validator.NewValidatorSet(epoch.Validators, c.logger)
 
-	exitRootHash, err := c.bridgeManager.BuildExitEventRoot(epoch.Number)
-	if err != nil {
-		return fmt.Errorf("could not build exit root hash for fsm: %w", err)
-	}
-
 	ff := &fsm{
-		config:              epoch.CurrentClientConfig,
-		forks:               c.config.Forks,
-		parent:              parent,
-		backend:             c.config.blockchain,
-		polybftBackend:      c.config.polybftBackend,
-		exitEventRootHash:   exitRootHash,
-		epochNumber:         epoch.Number,
-		blockBuilder:        blockBuilder,
-		validators:          valSet,
-		isEndOfEpoch:        isEndOfEpoch,
-		isEndOfSprint:       isEndOfSprint,
-		isFirstBlockOfEpoch: isFirstBlockOfEpoch,
-		proposerSnapshot:    proposerSnapshot,
-		logger:              c.logger.Named("fsm"),
+		config:                        epoch.CurrentClientConfig,
+		forks:                         c.config.Forks,
+		parent:                        parent,
+		backend:                       c.config.blockchain,
+		polybftBackend:                c.config.polybftBackend,
+		epochNumber:                   epoch.Number,
+		blockBuilder:                  blockBuilder,
+		validators:                    valSet,
+		isEndOfEpoch:                  isEndOfEpoch,
+		isEndOfSprint:                 isEndOfSprint,
+		isFirstBlockOfEpoch:           isFirstBlockOfEpoch,
+		proposerSnapshot:              proposerSnapshot,
+		logger:                        c.logger.Named("fsm"),
+		proposerBridgeBatchToRegister: make(map[uint64]*BridgeBatchSigned),
 	}
 
 	if isEndOfSprint {
-		commitment, err := c.bridgeManager.Commitment(pendingBlockNumber)
-		if err != nil {
-			return err
-		}
+		for chainID, bridgeManager := range c.bridgeManagers {
+			bridgeBatch, err := bridgeManager.BridgeBatch(pendingBlockNumber)
+			if err != nil {
+				return err
+			}
 
-		ff.proposerCommitmentToRegister = commitment
+			ff.proposerBridgeBatchToRegister[chainID] = bridgeBatch
```

Comment:
If its `nil` (that bridge manager does not have a batch built), do not add it to the map.